### DISCUSSION TITLE
Support many traversing behaviors

### DISCUSSION THREAD

**Discussion by ramnes** [qtile/qtile#2938] [Ideas]
Continuation of #1875.

Ideally we should have an API where we can do all this things when hitting a border:
* going to the opposite border of the current screen
* going to the opposite border of all screens
* going to another group
* doing nothing

For example, let's suppose I have two screens in line:

| 1 | 2 |
|---|---|

When hitting the left border of screen 1 with `traverse_left`, I may want one of the following:
* going to the right border of screen 1
* going to the right border of screen 2
* going to the previous group
* doing nothing

Where it gets complicated is that I probably want a different behavior on the edge of all screens than on a common border between two screens; I may want to do nothing on the left of screen 1, but jump to 2 on the right of 1... :)

Also, I think we should have the same methods available for dragging windows around, i.e. something like `window.cmd_move_{direction}`.

**Comment by tych0** [comment]
+1, I haven't looked at the patch in the PR closely, but in general I would like an API for this as well.
